src/App.vue

```
@@ -0,0 +1,251 @@
+      const newPost = await createPost({
+        ...postData,
+        userId: store.user.id,
+      });
+      store.posts.push(newPost);
+      store.selectedPost = newPost;
+      sidebarMode.value = 'view';
+      store.comments = [];
+    } else if (sidebarMode.value === 'edit') {
+      const updatedPost = await updatePost({
+        id: store.selectedPost.id,
+        ...postData,
+      });
+      const index = store.posts.findIndex((p) => p.id === updatedPost.id);
+      if (index !== -1) {
+        store.posts[index] = { ...store.posts[index], ...updatedPost };
+      }
+      store.selectedPost = { ...store.selectedPost, ...updatedPost };
+      sidebarMode.value = 'view';
+    }
+  } catch (error) {}
+};
+
+const handleEditPost = () => {
+  sidebarMode.value = 'edit';
+};
+
+const handleDeletePost = async () => {
+  try {
+    await deletePost(store.selectedPost.id);
+    store.posts = store.posts.filter((p) => p.id !== store.selectedPost.id);
+    store.selectedPost = null;
+    store.isSidebarOpen = false;
+    sidebarMode.value = '';
+  } catch (error) {}
+};
+
+const handleCloseSidebar = () => {
+  store.isSidebarOpen = false;
+  store.selectedPost = null;
+  sidebarMode.value = '';
+};
+
+const loadComments = async (postId) => {
+  store.areCommentsLoading = true;
+  store.commentsError = null;
+  try {
+    store.comments = await getComments(postId);
+  } catch (error) {
+    store.commentsError = 'Failed to load comments';
+  } finally {
+    store.areCommentsLoading = false;
+  }
+};
+
+const handleAddComment = async (commentData) => {
+  try {
+    const newComment = await createComment({
+      postId: store.selectedPost.id,
+      ...commentData,
+    });
+    store.comments.push(newComment);
+  } catch (error) {}
+};
+
+const handleDeleteComment = async (commentId) => {
+  store.comments = store.comments.filter((c) => c.id !== commentId);
+  
+  try {
+    await deleteComment(commentId);
+  } catch (error) {}
```

According to the requirements, 'Delete' errors must be handled. The comment is removed from the UI optimistically, but if the API call fails, the error is ignored. This will cause the comment to reappear on the next refresh, which is confusing for the user. You should handle the error, for instance, by showing a notification and restoring the comment to the list.